### PR TITLE
*: fix goword check, add it to Makefile, remove pre_commit hook

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Remove pre_commit check, and make it compulsory by adding `goword` check to the Makefile

### What is changed and how it works?

Fix goword check, and use it.

```make goword```

### Check List 

Tests 

 - No code

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @jackysp @zz-jason

File: ddl/index.go [hunk 1]
```diff
@@ -484,6 +484,7 @@ type addIndexWorker struct {
 	priority  int
 
 	// The following attributes are used to reduce memory allocation.
+
```
Comment (zz-jason) [inline ddl/index.go]: the empty line is generated by goword?

Comment (tiancaiamao) [reply ddl/index.go]: I add it... if there isn't an empty line, goword will complain that this comment should begin with `defaultVals`

**Comment by tiancaiamao** [comment]
PTAL @zz-jason

File: executor/merge_join.go [hunk 2]
```diff
@@ -77,7 +77,8 @@ type mergeJoinInnerTable struct {
 	joinKeys []*expression.Column
 	ctx      context.Context
 
-	// for chunk executions
+	// The following fields are used for chunk executions
```
[the remark this answers is not in the document]
Comment (tiancaiamao) [reply executor/merge_join.go]: Done

**Comment by tiancaiamao** [comment]
PTAL @XuHuaiyu

**Comment by tiancaiamao** [comment]
Some strategies to guarantee the code quality for our project, as you mentioned this work before.
PTAL @shenli

**Comment by jackysp** [comment]
PTAL @XuHuaiyu @zz-jason

File: ddl/index.go (hunk 1 above)
Comment (zz-jason) [reply ddl/index.go]: how about following the suggestion of goword?

Comment (tiancaiamao) [reply ddl/index.go]: What is the suggestion of goword?

**Comment by zz-jason** [comment]
@tiancaiamao sorry for the late review, please update to the latest master branch and resolve conflicts.

**Pull request closed without merging**